### PR TITLE
Slurm 21.08 Upgrade in 2.11

### PR DESCRIPTION
**Pull request by Nbot07A**
### Description of changes
* Describe *what* you're changing and *why* you're doing these changes.
Upgrading slurm to 21.08.8-2 
Emulating https://github.com/aws/aws-parallelcluster/pull/3452#issue-1048182485
* Link to impacted open issues.
N/A
### Tests
* Describe the automated and/or manual tests executed to validate the patch.
passed `test_slurm` integration tests
* Describe the added/modified tests.
N/A

### References
* Link to related PRs in other packages (i.e. cookbook, node).
These PRs are also part of the upgrade:
https://github.com/aws/aws-parallelcluster-node/pull/431
https://github.com/aws/aws-parallelcluster-cookbook/pull/1500
* Link to documentation useful to understand the changes.
https://www.schedmd.com/news.php
https://github.com/aws/aws-parallelcluster-cookbook/pull/1193
https://github.com/aws/aws-parallelcluster-node/pull/350/files
https://github.com/aws/aws-parallelcluster/pull/3452
 https://github.com/aws/aws-parallelcluster-cookbook/pull/1430
 https://github.com/aws/aws-parallelcluster-node/pull/302/commits/cc3bad5ebb303c93608878676a0ae6a6563f6662#diff-37d18a99e1edd3058be1192c765dcff454f9891b65cc21622200d59008e58169
https://github.com/aws/aws-parallelcluster-node/blob/develop/src/slurm_plugin/slurm_resources.py#L77
https://github.com/aws/aws-parallelcluster-node/blob/release-2.11/src/common/schedulers/slurm_commands.py#L49

### Checklist
- [x] Make sure you are pointing to **the right branch** and add a label in the PR title (i.e. **2.x** vs **3.x**)
- [ ] Check all commits' messages are clear, describing what and why vs how.
- [ ] Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- [ ] Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,12 @@ CHANGELOG
 **CHANGES**
 Upgrade Python runtime used by Lambda functions in AWS Batch integration to python3.9.
 
+2.11.8
+-----
+
+**CHANGES**
+- Upgrade Slurm to version 21.08.8-2
```
Comment (hanwen-pcluste) [inline CHANGELOG.md]: Missing final period.

**Comment by hanwen-pcluste** [comment]
Great work!

Please take a look at the PR description of https://github.com/aws/aws-parallelcluster/pull/3452 and copy them to the `Description of changes` of this PR description.

Moreover, under `Tests`, you can say the `test_slurm` integration tests have been passed

**Comment by hanwen-pcluste** [comment]
Nice to have (optional): We can improve the `test_slurm` to cluster cluster update as https://github.com/aws/aws-parallelcluster/pull/3452/files#diff-5d2b210a3df82d89d8bed3f1d7d0bc48e5998e3eb9358c07aaa75c5df2d5d44eR1002

**Review by demartinofra — COMMENTED** [review]
* Please also backport `test_scontrol_reboot` in integration tests and have it enabled in the test config.
* Let's change the destination branch and review/squash commit messages before merging
* Also could you check what are the diffs with the slurm integ tests in develop and see if anything else needs to be aligned. You could limit the diff to slurm tests that are already defined in the release-2.11 branch

**Review by chenwany — APPROVED** [review]

**Pull request merged**